packages/scene-composer/src/components/panels/scene-components/tag-style/ColorPicker/ColorPicker.tsx

```diff
@@ -69,22 +86,62 @@ export const ColorPicker = ({ color, onSelectColor, label }: IColorPickerProps):
     setShowChromePicker(false);
   };
 
-  const handleCloseChromePicker = () => {
+  const handleCloseCustomPicker = () => {
     setShowChromePicker(false);
   };
 
+  const handleColorChange = useCallback(
+    (color: ColorResult) => {
+      checkIfCustomColor(color.hex);
+      setHexCodeError(''); // Clear any existing error message
+      setNewColor(color.hex);
+      onSelectColor(color.hex);
+    },
+    [color, onSelectColor, onUpdateCustomColors],
+  );
+
+  const handleHexCodeChange = useCallback(
+    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
+      setNewColor(event.detail.value);
```

Comment:
Do we want to set the color if it's not valid hex code?

Reply:
First, we want to set it to the state variable as this is coming as external prop. Then we are validating whether it is a valid color or not. In the same method we are validating.

Reply:
in handleHexCodeChange it doesn't look like an external prop yet. it's an internal event.detail.value that we haven't validated yet?

Reply:
const [newColor, setNewColor] = useState<string>(color);
Color is a prop and I am using a single state variable to update the state of the component. So initially will have a value that is coming from prop. If user change the hex code then we are checking whther its a valid or not then resetting it again.
const handleHexCodeChange = useCallback(
    (event: NonCancelableCustomEvent<InputProps.ChangeDetail>) => {
      setNewColor(event.detail.value);
}